src/test/java/gov/usgs/wma/waterdata/groundwater/RdbWriterTest.java

```diff
@@ -100,40 +106,101 @@ void testHeaderWriten() {
+					break;
+				case "lev_tm":
+					assertTrue(val.matches("\\d{4}"), "lev_dt should match 'HHmm'");
+					break;
+				case "lev_ent_cd":
+					assertEquals("S", val);
+					break;
+				case "lev_md":
+					assertEquals("07-MAY-2007 18:30:47", val);
+					break;
+				case "lev_dtm":
+					assertEquals("01-MAY-2007 12:00:00", val);
+					break;
+				default:
+					assertEquals("", val, "'" + val + "' should be an empty string (column index " + i + ", " + colName + ")");
+			}
+
+		}
 	}
 
 	@Test
 	void testRowWriteGeneral() {
 		// SETUP
 		DiscreteGroundWater dgw = makeDgw();
+		rdbWriter.writeHeader();
 
 		// ACTION UNDER TEST
 		rdbWriter.writeRow(dgw);
 
 		// POST SETUP
-		long count = rdbWriter.close().getDataRows();
-		String writtenValue = out.toString();
+		long count = rdbWriter.getDataRowCount();
+		String[] writtenValues = rdbWriter.getDataRowsContent();
+		String writtenValue = writtenValues[0];
 
 		// ASSERTIONS
-		assertNotNull(writtenValue);
-		assertTrue(writtenValue.startsWith("USGS\t")); // first USGS value
-		assertTrue(writtenValue.contains("\tUSGS\t")); // second USGS value
-		assertTrue(writtenValue.contains("\t4042342342\t"));
-		assertTrue(writtenValue.contains("\t2\t"));
-		assertTrue(writtenValue.contains("\tS\t"));
-		assertTrue(writtenValue.contains("\tT\t"));
-		assertTrue(writtenValue.contains("\t200705"));
-		assertTrue(writtenValue.contains("\t20070501\t"));
-		assertTrue(writtenValue.contains("\t1200\t"));
-		assertFalse(writtenValue.contains("\t1830\t"));
-		assertTrue(writtenValue.contains("\t07-MAY-2007 18:30:47\t"));
-		assertTrue(writtenValue.contains("\t01-MAY-2007 12:00:00\t")); // measured and UTC dates
-		assertTrue(writtenValue.contains("\tUTC\t"));
-		assertTrue(writtenValue.contains("\tD\t"));
-		assertTrue(writtenValue.contains("\t30210")); // last value
-
-		assertEquals(1, count);
+		assertEquals(1, writtenValues.length);
+		assertTrue(writtenValues[0].startsWith("USGS\t"));  //Sanity check that the string actually starts as expected
+
+		assertEquals("USGS", rdbWriter.getLastValueFor("agency_cd"));
+		assertEquals("4042342342", rdbWriter.getLastValueFor("site_no"));
+		assertEquals("2", rdbWriter.getLastValueFor("lev_acy_cd"));
+		assertEquals("L", rdbWriter.getLastValueFor("lev_ent_cd"));
+		assertEquals("USGS", rdbWriter.getLastValueFor("lev_agency_cd"));
+		assertEquals("S", rdbWriter.getLastValueFor("lev_meth_cd"));
+		assertEquals("T", rdbWriter.getLastValueFor("lev_age_cd"));
+		assertEquals("20070501", rdbWriter.getLastValueFor("lev_dt"));
+		assertEquals("1200", rdbWriter.getLastValueFor("lev_tm"));
+		assertEquals("01-MAY-2007 12:00:00", rdbWriter.getLastValueFor("lev_dtm"));
+		assertEquals("07-MAY-2007 18:30:47", rdbWriter.getLastValueFor("lev_md"));
+		assertEquals("UTC", rdbWriter.getLastValueFor("lev_tz_cd"));
+		assertEquals("D", rdbWriter.getLastValueFor("lev_dt_acy_cd"));
+		assertEquals("30210", rdbWriter.getLastValueFor("parameter_code"));
```

Comment:
I needed to be more specific when investigating this, so I expanded the mock rdbWriter to be able to match up the column names and values.